### PR TITLE
using formik

### PR DESCRIPTION
**Pull request by manitheleo**

### CODE REVIEW

File: src/Components/ToggleEdit.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
     );
   } else {
-    return (
-      <Form
-        handleFormSubmit={props.handleFormSubmit}
-        todo={props.todo}
-        handleInputChange={props.handleInputChange}
-      />
-    );
+    return <Form formik={formik} />;
```
Comment (ArashKianiamphora) [inline src/Components/ToggleEdit.js]: why not just pass in handleSubmit and getFieldProps ?

File: src/Components/ToggleEdit.js [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
 import Form from "./Forms/Form";
-const ToggleEdit = (props) => {
-  if (props.edit) {
+const ToggleEdit = ({
+  formik,
+  edit,
+  handleEditFormSubmit,
+  setEdit,
+  currentTodo,
+  handleEditInput,
```
Comment (ArashKianiamphora) [inline src/Components/ToggleEdit.js]: well done 🙌

File: src/Components/ListsItem.js [hunk 3]
```diff
@@ -1,16 +1,17 @@
 import { List } from "@mui/material";
 
 import Listitem from "./Listitem";
-const ListsItem = (props) => {
+const ListsItem = ({ tasks, handleEditClick, handleDeleteItem }) => {
+  console.log(tasks);
```
Comment (ArashKianiamphora) [inline src/Components/ListsItem.js]: always check your code. so there is no console.log

Comment (manitheleo) [reply src/Components/ListsItem.js]: sorry sir i will change it

File: src/Components/ToggleEdit.js [hunk 4]
```diff
@@ -1,23 +1,24 @@
 import EditForm from "./Forms/EditForm";
 import Form from "./Forms/Form";
```
Comment (ArashKianiamphora) [inline src/Components/ToggleEdit.js]: you named it FormSubmit and exported the default and using it with the name Form, when coding its important to be consistent and descriptive. try to name your file, component and use it with the same name

Comment (manitheleo) [reply src/Components/ToggleEdit.js]: ok sir

File: src/Components/ToggleEdit.js (hunk 1 above)
Comment (manitheleo) [reply src/Components/ToggleEdit.js]: it is easily understand that why i didn't pass sir

Comment (ArashKianiamphora) [reply src/Components/ToggleEdit.js]: it's always better to pass on what you need, please change as per my suggestion.

File: src/Components/Forms/Form.js [hunk 5]
```diff
@@ -1,14 +1,14 @@
 import { Button, InputAdornment, TextField } from "@mui/material";
-
-const FormSubmit = (props) => {
+// import {useFormik} from 'formik'
```
Comment (ArashKianiamphora) [inline src/Components/Forms/Form.js]: why are we adding this exactly

File: src/Components/Forms/Form.js [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+const FormSubmit = ({ formik }) => {
```
Comment (ArashKianiamphora) [inline src/Components/Forms/Form.js]: please only import `handleSubmit` and `getFieldProps`

File: src/Components/ToggleEdit.js [hunk 7]
```diff
@@ -1,23 +1,24 @@
 import EditForm from "./Forms/EditForm";
-import Form from "./Forms/Form";
-const ToggleEdit = (props) => {
-  if (props.edit) {
+import FormSubmit from "./Forms/FormSubmit";
```
Comment (ArashKianiamphora) [inline src/Components/ToggleEdit.js:2]: much better, now you can search for it easily.

**Review by ArashKianiamphora — CHANGES_REQUESTED** [review]

File: src/Components/Forms/Form.js (hunk 6 above)
Comment (manitheleo) [reply src/Components/Forms/Form.js]: ok Sir

**Review by ArashKianiamphora — APPROVED** [review]

**Pull request merged**